compiler/rustc_mir/src/transform/large_enums.rs

```diff
@@ -0,0 +1,218 @@
+                                Rvalue::Use(Operand::Copy(Place {
+                                    local: discr_place.local,
+                                    projection: tcx
+                                        .intern_place_elems(&[PlaceElem::Index(size_place.local)]),
+                                })),
+                            )),
+                        };
+
+                        // FIXME(jknodt) do I need to add a storage live here for this place?
+                        let dst = Place {
+                            local: patch.new_temp(
+                                tcx.mk_ty(ty::RawPtr(ty::TypeAndMut {
+                                    ty,
+                                    mutbl: Mutability::Mut,
+                                })),
+                                span,
+                            ),
+                            projection: List::empty(),
+                        };
+
+                        let dst_ptr = Statement {
+                            source_info,
+                            kind: StatementKind::Assign(box (
+                                dst,
+                                Rvalue::AddressOf(Mutability::Mut, *lhs),
+                            )),
+                        };
+
+                        // FIXME(jknodt) do I need to add a storage live here for this place?
+                        let src = Place {
+                            local: patch.new_temp(
+                                tcx.mk_ty(ty::RawPtr(ty::TypeAndMut {
+                                    ty: rhs.ty(local_decls, tcx).ty,
+                                    mutbl: Mutability::Mut,
+                                })),
+                                span,
+                            ),
+                            projection: List::empty(),
+                        };
+
+                        let src_ptr = Statement {
+                            source_info,
+                            kind: StatementKind::Assign(box (
+                                src,
+                                Rvalue::AddressOf(Mutability::Mut, *rhs),
+                            )),
+                        };
+
+                        let copy_bytes = Statement {
+                            source_info,
+                            kind: StatementKind::CopyNonOverlapping(box CopyNonOverlapping {
+                                src: Operand::Copy(src),
+                                dst: Operand::Copy(src),
+                                count: Operand::Constant(
+                                    box (Constant {
+                                        span,
+                                        user_ty: None,
+                                        literal: ConstantKind::Val(
+                                            interpret::ConstValue::from_u64(1),
+                                            tcx.types.usize,
+                                        ),
```

Comment:
I would expect this to be a size in bytes with src & dst pointers casted to say `*const u8` and `*mut u8`.

Reply:
This size is the number of elements, and since they should be the same type I think having one element should work. I can also make it u8s but I don't know if there's any difference

Reply:
Using an enum type and one element would copy the whole enum. The goal is to copy the active variant only.